### ISSUE TITLE
`.io` ccTLD looks like it's heading for retirement - long term plan for `crates.io`?

### ISSUE THREAD

**Issue by cyqsimon** [rust-lang/crates.io#9864]
## TLDR

As you the reader probably know, the `.io` ccTLD is scheduled to be "phased out" in the next couple of years, whatever "phased out" means in this case. In the best case scenario, `crates.io`'s continued existence would be dependent on IANA's gratuity; in the worst case, a sudden disappearance of `crates.io` would be disastrous for the entire Rust ecosystem. I think it's a risk that should be discussed and dealt with sooner than later.

## What will happen to `.io`?

The answer is nobody knows. Historically, retired ccTLDs due to various geopolitical issues have mostly been removed. Here's a short list:

| ccTLD | Country        | Deprecation | Deletion | Note                   |
|-------|----------------|-------------|----------|------------------------|
| `.dd` | East Germany   | N/A         | N/A      | Never added to root NS |
| `.cs` | Czechoslovakia | 1993        | 1995     |                        |
| `.yu` | Yugoslavia     | 2003        | 2010     |                        |
| `.su` | Soviet Union   | 1994        | N/A      | Retained by Russia     |
| `.tp` | East Timor     | 2005        | 2015     |                        |

It's arguable that `.io` is similar to `.su` in terms of its wide usage, so IANA may end up making an exception. Although it has to be said, IANA has attempted a move to remove `.su` entirely, and it took significant convincing by the Russian government for it to end up retained. `.io` may have a lot of corporate interest backing it too, but it's no nation state. Regardless, I think we can agree this is a significant risk, the outcome of which is beyond any of our control.

## Impact to `crates.io`

It's likely that nothing horrible is going to happen in the near term, given how slowly such processes progress. However practical issues may start to arise. For example, when Brexit happened, [British citizens were no longer able to register or renew their `.eu` domains in a surprisingly short amount of time](https://www.gov.uk/guidance/registering-and-renewing-eu-domain-names-in-the-uk).

And again, it's good to have some sort of long term plan regardless.

**Comment by Turbo87** [comment]
see https://github.com/rust-lang/crates.io/discussions/9634

**Comment by cyqsimon** [comment]
> see [#9634](https://github.com/rust-lang/crates.io/discussions/9634)

Ah okay. As long as it's being talked about.

I did a search in issues, but I forgot discussions is a thing. Sorry about that.

**Comment by Alonely0** [comment]
If `.io` does indeed get removed, the most straightforward—yet cursed—thing we can do is host it under `crates.io.rust-lang.org`; alternatively, we could rebrand `crates.io` to just `crates` (which is rather problematic, and the term becomes way vaguer), resulting in something more sane like `crates.rust-lang.org`. Either way, it's better to do it earlier rather than later, and once a course of action has been set, make the `crates.io` domain redirect to wherever we end up hosting; we don't really want to start moving domains just months before IANA kills `.io`, so even if it's in a gray area right now, taking action (or at least deciding which should be taken) is preferable.
